clang/test/dpct/kernel_without_name.cu

```diff
@@ -277,3 +277,34 @@ void run_foo7(T *a, const T *b, const T *c, const T *d, const T *e, const int f,
     foo_kernel6<<<grid, block, 0, stream>>>(a, b, c, d, e, f, g, h);
   }
 }
+
+#ifndef BUILD_TEST
+template <typename T> struct kernel_type_t {
+  using Type = T;
+};
+
+//CHECK:template <typename T> void foo_kernel7(Tk *mem) {
+//CHECK-NEXT:  /*
```

Comment:
Move warning to Line 286 and kernel call location, and tell user the Tk is defined in function foo_kernel7